src/backend/distributed/transaction/backend_data.c

```diff
@@ -0,0 +1,480 @@
+
+	distributedTransctionId = GetCurrentDistributedTransctionId();
+
+	memset(values, 0, sizeof(values));
+	memset(isNulls, false, sizeof(isNulls));
+
+	/* first two fields do not change for this backend, so get directly */
+	values[0] = ObjectIdGetDatum(MyDatabaseId);
+	values[1] = Int32GetDatum(MyProcPid);
+
+	values[2] = Int32GetDatum(distributedTransctionId->initiatorNodeIdentifier);
+	values[3] = UInt64GetDatum(distributedTransctionId->transactionNumber);
+	values[4] = TimestampTzGetDatum(distributedTransctionId->timestamp);
+
+	heapTuple = heap_form_tuple(tupleDescriptor, values, isNulls);
+
+	PG_RETURN_DATUM(HeapTupleGetDatum(heapTuple));
+}
+
+
+/*
+ * get_all_active_transactions returns all the avaliable information about all
+ * the active backends.
+ */
+Datum
+get_all_active_transactions(PG_FUNCTION_ARGS)
+{
+	ReturnSetInfo *returnSetInfo = (ReturnSetInfo *) fcinfo->resultinfo;
+	TupleDesc tupleDescriptor = NULL;
+	Tuplestorestate *tupleStore = NULL;
+	MemoryContext perQueryContext = NULL;
+	MemoryContext oldContext = NULL;
+
+	int backendIndex = 0;
+
+	const int attributeCount = 5;
+	Datum values[attributeCount];
+	bool isNulls[attributeCount];
+
+	CheckCitusVersion(ERROR);
+
+	/* check to see if caller supports us returning a tuplestore */
+	if (returnSetInfo == NULL || !IsA(returnSetInfo, ReturnSetInfo))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("set-valued function called in context " \
+						"that cannot accept a set")));
+	}
+
+	if (!(returnSetInfo->allowedModes & SFRM_Materialize))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("materialize mode required, but it is not " \
+						"allowed in this context")));
+	}
+
+	/* build a tuple descriptor for our result type */
+	if (get_call_result_type(fcinfo, NULL, &tupleDescriptor) != TYPEFUNC_COMPOSITE)
+	{
+		elog(ERROR, "return type must be a row type");
+	}
+
+	perQueryContext = returnSetInfo->econtext->ecxt_per_query_memory;
+
+	oldContext = MemoryContextSwitchTo(perQueryContext);
+
+	tupleStore = tuplestore_begin_heap(true, false, work_mem);
+	returnSetInfo->returnMode = SFRM_Materialize;
+	returnSetInfo->setResult = tupleStore;
+	returnSetInfo->setDesc = tupleDescriptor;
+
+	MemoryContextSwitchTo(oldContext);
+
+	/*
+	 * We don't want to initialize memory while spinlock is held so we
+	 * prefer to do it here. This initialization is done only for the first
+	 * row.
+	 */
+	memset(values, 0, sizeof(values));
+	memset(isNulls, false, sizeof(isNulls));
+
+	/* we're reading all the backend data, take a lock to prevent concurrent additions */
+	LWLockAcquire(AddinShmemInitLock, LW_SHARED);
+
+	for (backendIndex = 0; backendIndex < MaxBackends; ++backendIndex)
+	{
+		BackendData *currentBackend =
+			&backendManagementShmemData->backends[backendIndex];
+
+		SpinLockAcquire(&currentBackend->mutex);
+
+		/* we're only interested in active backends */
+		if (currentBackend->transactionId.transactionNumber == 0)
```

Comment:
For readability and prettiness, you could consider adding `IsInDistributedTransaction` to this PR:
https://github.com/citusdata/citus/pull/1495/files#diff-330566a321cdbe3ff78b455379575c5cR743